src/lib/config.js

```diff
@@ -92,66 +117,100 @@ async function config() {
     return state.cancelled ? red(state.symbols.cross) : state.symbols.check;
   }
 
-  const response = await prompt([
-    {
-      type: 'input',
-      name: 'deployAliasName',
-      message: (state) => {
-        const style = state.submitted && !state.cancelled ? green : reset;
-        return style('Choose a name (can be anything):');
-      },
-      prefix: formatPrefixSymbol,
-      validate: async (val) => {
-        val = val.toLowerCase().trim().replace(' ', '-');
-        if (!val) return red('Name is required.');
-        if (Object.keys(config.deployAliases).includes(val)) {
-          return red('Name already exists.');
-        }
-        return true;
-      },
-      result: (val) => val.toLowerCase().trim().replace(' ', '-'),
-    },
-    {
-      type: 'input',
-      name: 'url',
-      message: (state) => {
-        const style = state.submitted && !state.cancelled ? green : reset;
-        return style('Set the Mina GraphQL API URL to deploy to:');
-      },
-      prefix: formatPrefixSymbol,
-      validate: (val) => {
-        if (!val) return red('Url is required.');
-        return true;
-      },
-      result: (val) => val.trim().replace(/ /, ''),
-    },
-    {
-      type: 'input',
-      name: 'fee',
-      message: (state) => {
-        const style = state.submitted && !state.cancelled ? green : reset;
-        return style('Set transaction fee to use when deploying (in MINA):');
-      },
-      prefix: formatPrefixSymbol,
-      validate: (val) => {
-        if (!val) return red('Fee is required.');
-        if (isNaN(val)) return red('Fee must be a number.');
-        if (val < 0) return red("Fee can't be negative.");
-        return true;
-      },
-      result: (val) => val.trim().replace(/ /, ''),
-    },
+  const {
+    deployAliasPrompts,
+    initialFeepayerPrompts,
+    recoverFeepayerPrompts,
+    otherFeepayerPrompts,
+    feepayerAliasPrompt,
+  } = prompts;
+
+  const initialPromptResponse = await prompt([
```

Comment:
I chose this approach of feeding the response of the initial prompts into the subsequent branching prompt options because there is not a good api and way to handle nested prompts that depend on previous answers with the package we are using in the cli.